src/Main.js

```diff
@@ -7,6 +7,10 @@ const Logger = Log.Logger;
 
 const WebhookHandler = createHandler({ path: Config.githubWebhookPath(), secret: Config.githubWebhookSecret() });
 
+process.on('unhandledRejection', error => {
+    Logger.error("unhandledRejection", error.message);
```

Comment:
I think we should quit/abort the whole Anubis process on these errors. Would that be difficult?

Reply:
Done.

Reply:
As discussed during the meeting, please check whether process.exit() is the best way to immediately stop the program execution.

Reply:
The [documentation](https://nodejs.org/api/process.html#process_process_exit_code) says that it "forces the process to exit as quickly as possible even if there are still asynchronous operations pending that have not yet completed fully". I do not see any problems with it. Quick googling did not show such potential problems either.

Reply:
In my tests I have not noticed any differences: in both cases, the program terminates and logs are saved. However, if the docs say that rethrowing is probably more reliable (that logs may be dropped sometimes due to process.exit()), we should use the rethrow approach. I found another related [discussion](https://stackoverflow.com/questions/5266152/how-to-exit-in-node-js), where some people recommend using this approach either.
I pushed that change at 0141804.

Reply:
Another case, which is uncaughtException should be covered also. In my tests, I see problems with logging in this handler: the logs are printed into stderr/stdout, but the log file still lacks them There is a general problem, discussed [here](https://github.com/trentm/node-bunyan/issues/37). The possible solution, using [exeunt](https://github.com/joyent/node-exeunt) module, works only for stderr/stdout, and does not work for files.

Reply:
I do not think we should risk post-exception execution of event handlers or spend time on trying to prevent that all such executions except for logging event handlers. It feels like we would be solving an unsolvable problem.

Instead, to retain crash information, I suggest that we use [node --abort-on-unhandled-exception](https://nodejs.org/dist/v8.9.3/docs/api/cli.html#cli_abort_on_uncaught_exception) approach as [mentioned](https://github.com/trentm/node-bunyan/issues/37#issuecomment-298810505) on the Bunyan thread you found. AFAICT, that approach requires node v0.10. Are we already using that (or later) node version? If not, we can wait until we upgrade.

If `--abort-on-unhandled-exception` does not prevent us from using a custom uncaught exception handler, then, in _addition_ to using that node option, please install that handler to log a brief, safe message about dying on an uncaught exception to stderr, even though that message is likely to be lost when Anubis is started from a system script (as it should be).

Reply:
> Are we already using that (or later) node version?

That is a rather old version, we use now v9.0.0 or later.

> If --abort-on-unhandled-exception does not prevent us from using a custom uncaught exception handler,
With this option, the custom `uncaughtException` handler is not called

I restarted the bot with `--abort-on-unhandled-exception` on the server, and now we can only notice a crash by core files generated. How should we limit core file size there?

Reply:
> we can only notice a crash by core files generated

Is it fairly easy to get usable crash information (e.g., which exception was not caught and where it was thrown) from the core file?

> How should we limit core file size there?

I am not a sysadmin, and I do not know about that specific server policies. Can we just use whatever setting are already configured there? Or are the core dumps disabled on that server right now? At any rate, please see if you and kinkie (who, I presume, administers that server) can agree on a reasonable setting, whatever it is (e.g., unlimited size with the last few preserved).

Reply:
> Is it fairly easy to get usable crash information

I found an [article](https://www.javascriptjanuary.com/blog/nodejs-postmortem-debugging-for-fun-and-production) describing such debugging with lldb + llnode. Looks not hard, but I have not experimented with it (which can be done when required).

> Can we just use whatever setting are already configured there?

I tried to force the bot to generate core file and it worked there. So it looks like no ulimit adjustments needed there.